### PR TITLE
refactor(flutter-deriv-api): Remove android ios modules

### PR DESCRIPTION
**Pull request by abedelaziz-deriv**
### What was done?

- Android and ios modules were removed; converting the plugin into a dart package.
- Updated the branch with the changes of master.
- Fixed tests and removed un-useful tests.
- Updated the example app's dependencies.

### CODE REVIEW

File: example/android/settings.gradle [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
-    def pluginDirectory = flutterProjectRoot.resolve(path).resolve('android').toFile()
-    include ":$name"
-    project(":$name").projectDir = pluginDirectory
+plugins {
+    id "dev.flutter.flutter-plugin-loader" version "1.0.0"
+    id "com.android.application" version '8.5.0' apply false
+    id "org.jetbrains.kotlin.android" version "1.9.20" apply false
 }
+
+include ':app'
```
Comment (ramin-deriv) [inline example/android/settings.gradle:25]: would you please fix EOF.

Comment (abedelaziz-deriv) [reply example/android/settings.gradle]: done

**Comment by ahrar-deriv** [comment]
i noticed we cant use flutter version 3.10.2 as the dart version is lower than the required one. no issue as we are upgrading the flutter version.

File: pubspec.yaml
Comment (ahrar-deriv) [inline pubspec.yaml:1]: flutter dependency can be removed. as we want to just use dart.
flutter: "3.10.2"
any flutter SDK can be removed from the yaml

Comment (abedelaziz-deriv) [reply pubspec.yaml:1]: done

**Review by ramin-deriv — APPROVED** [review]

**Pull request merged**